tests/integration/test_kubeflow_training.py

```diff
@@ -36,87 +62,125 @@ def train_fn_with_gcs_access():
+        gcs_prefix=temp_gcs_prefix,
+        capsys=capsys,
+        expected_result=GCS_SUCCESS_MSG)
 
-    captured = capsys.readouterr()
-    assert 'testing gcs access' in captured.out
-
-def test_tfjob_submission_with_gcs_access(capsys):
+def test_tfjob_submission_with_gcs_access(capsys, temp_gcs_prefix):
     run_submission_with_gcs_access(
         'tfjob',
         pod_spec_mutators=[fairing.cloud.gcp.add_gcp_credentials],
-        namespace='kubeflow')
+        namespace='kubeflow',
+        gcs_prefix=temp_gcs_prefix,
+        capsys=capsys,
+        expected_result=GCS_SUCCESS_MSG)
 
-    captured = capsys.readouterr()
-    assert 'testing gcs access' in captured.out
-
-def test_job_submission_without_gcs_access(capsys):
+def test_job_submission_without_gcs_access(capsys, temp_gcs_prefix):
     run_submission_with_gcs_access(
         'job',
         pod_spec_mutators=[],
-        namespace='kubeflow')
-
-    captured = capsys.readouterr()
-    assert 'google.api_core.exceptions.Forbidden: 403' in captured.out
+        namespace='kubeflow',
+        gcs_prefix=temp_gcs_prefix,
+        capsys=capsys,
+        expected_result=GCS_FAILED_MSG)
 
-def test_tfjob_submission_without_gcs_access(capsys):
+def test_tfjob_submission_without_gcs_access(capsys, temp_gcs_prefix):
     run_submission_with_gcs_access(
         'tfjob',
         pod_spec_mutators=[],
-        namespace='kubeflow')
-
-    captured = capsys.readouterr()
-    assert 'google.api_core.exceptions.Forbidden: 403' in captured.out
+        namespace='kubeflow',
+        gcs_prefix=temp_gcs_prefix,
+        capsys=capsys,
+        expected_result=GCS_FAILED_MSG)
 
-def test_job_submission_invalid_namespace(capsys):
+def test_job_submission_invalid_namespace(capsys, temp_gcs_prefix):
     with pytest.raises(ValueError) as err:
         run_submission_with_gcs_access(
             'job',
             pod_spec_mutators=[fairing.cloud.gcp.add_gcp_credentials],
-            namespace='default')
+            namespace='default',
+            gcs_prefix=temp_gcs_prefix,
+            capsys=capsys,
+            expected_result=None)
 
     msg = 'Unable to mount credentials: '\
           'Secret user-gcp-sa not found in namespace default'
     assert msg in str(err.value)
 
-def test_tfjob_submission_invalid_namespace(capsys):
+def test_tfjob_submission_invalid_namespace(capsys, temp_gcs_prefix):
     with pytest.raises(ValueError) as err:
         run_submission_with_gcs_access(
             'tfjob',
             pod_spec_mutators=[fairing.cloud.gcp.add_gcp_credentials],
-            namespace='default')
+            namespace='default',
+            gcs_prefix=temp_gcs_prefix,
+            capsys=capsys,
+            expected_result=None)
 
     msg = 'Unable to mount credentials: '\
           'Secret user-gcp-sa not found in namespace default'
     assert msg in str(err.value)
 
+def run_submission_with_high_level_api(backend, entry_point, capsys, expected_result):
```

Comment:
The high level api tests should be in a separate file, especially as they'll get more complex over time.

Reply:
done.